core/trino-main/src/main/java/io/trino/Session.java

```diff
@@ -129,8 +129,8 @@ public Session(
         this.identity = requireNonNull(identity, "identity is null");
         this.originalIdentity = requireNonNull(originalIdentity, "originalIdentity is null");
         this.source = requireNonNull(source, "source is null");
-        this.catalog = requireNonNull(catalog, "catalog is null");
-        this.schema = requireNonNull(schema, "schema is null");
+        this.catalog = requireNonNull(catalog, "catalog is null").map(value -> value.toLowerCase(Locale.ENGLISH));
```

Comment:
I think that sanitization should happen in the `HttpRequestSessionContextFactory`

Reply:
Catalog and schema are read from the headers so it should happen on read